src/lib/containers/entity/annotations/CustomObjectFieldTemplate.tsx

```diff
@@ -11,9 +13,62 @@ import AddToList from '../../../assets/icons/AddToList'
  * @param props
  * @returns
  */
-export function CustomObjectFieldTemplate(props: ObjectFieldTemplateProps) {
+export function CustomObjectFieldTemplate(
+  props: ObjectFieldTemplateProps<Record<string, unknown>>,
+) {
   const { TitleField, DescriptionField } = props
   const CUSTOM_OBJECT_FIELD_TEMPLATE_TOOLTIP_ID = `CustomObjectFieldTooltip-${props.idSchema.$id}`
+
+  const [
+    previousSchemaDefinedProperties,
+    setPreviousSchemaDefinedProperties,
+  ] = useState<Set<string>>(new Set())
+
+  /**
+   * We track how the schema changes as the user enters data, causing conditional subschemas to be evaluated.
+   *
+   * If a property exists in the previous version of the schema and has user data, and then is dropped from the schema due to a data update, then
+   * we need to prompt the user whether they want to undo the change, or continue and keep/remove those fields
+   *
+   * In this component, we can identify when one or more fields with user data are dropped (the field gains the additional property flag in the schema prop).
+   * We then use a function provided by the context to report which fields were lost.
+   */
+  useDeepCompareEffectNoCheck(() => {
+    if (props.schema.properties) {
+      const propertyKeys = Object.keys(props.schema.properties)
+      // Schema-defined properties are those properties in the schema without the additional property flag.
```

Comment:
And this ADDITIONAL_PROPERTY_FLAG is our workaround for additional "custom" annotations (those annotations that do not conform to the schema)?

Reply:
The flag is applied and used by react-jsonschema-form internally to denote which properties are not defined by the schema. We can inspect it here since this is a RJSF template component.

So yes, this is a workaround in the sense that using this means we don't have to duplicate the work of computing the schema without conditionals (and also saves us from writing the code to do that correctly).